src/io/flutter/console/FlutterConsoleFolding.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2016 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.console;
+
+import com.intellij.execution.ConsoleFolding;
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.configurations.CommandLineTokenizer;
+import com.intellij.openapi.util.text.StringUtil;
+import io.flutter.sdk.FlutterSdk;
+import io.flutter.sdk.FlutterSdkUtil;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+public class FlutterConsoleFolding extends ConsoleFolding {
+  @Override
+  public boolean shouldFoldLine(String line) {
+    if (!line.contains("flutter run")) return false;
+
+    try {
+      final String homePath = FlutterSdk.getGlobalFlutterSdk().getHomePath();
```

Comment:
You might consider guarding against a null SDK here.  (Pretty sure an inspection would nag too since I bet `getGlobalFlutterSdk()` returns `@Nullable`.